### PR TITLE
Add clockwise check for polygon

### PR DESCRIPTION
**Pull request by matthew-larson**
This pull request creates a bool function checking whether the polygon is counterclockwise, otherwise it is modified to satisfy this requirement.  A unit test was added for this condition.  The document also reformatted for clang.

### CODE REVIEW

File: test/unit/foundation.unit.cpp [hunk 1]
```diff
@@ -285,50 +284,67 @@ TEST_F(TypicalFixture, convectionCallback) {
...
+  intIns.width = -0.05;
+  intIns.material = insulation;
+  fnd.inputBlocks.push_back(intIns);
+  fnd.polygon.clear();
+  fnd.polygon.outer().push_back(Point(-6.0, -6.0));
+  fnd.polygon.outer().push_back(Point(6.0, -6.0));
+  fnd.polygon.outer().push_back(Point(6.0, 6.0));
+  fnd.polygon.outer().push_back(Point(-6.0, 6.0));
+  fnd.createMeshData();
+}
```
Comment (nealkruis) [inline test/unit/foundation.unit.cpp:376]: What do we expect here? There are no tests (e.g., `EXPECT_TRUE()`).

**Review by nealkruis — CHANGES_REQUESTED** [review]

File: src/libkiva/Foundation.cpp [hunk 2]
```diff
@@ -132,6 +132,10 @@ void Foundation::createMeshData() {
     buildingSurfaces.push_back(poly);
   }
 
+  if (!isCounterClockWise(polygon)) {
+      boost::geometry::correct(polygon);
```
Comment (nealkruis) [inline src/libkiva/Foundation.cpp]: We should issue a warning here.

**Review by nealkruis — CHANGES_REQUESTED** [review]

File: test/unit/foundation.unit.cpp (hunk 1 above)
Comment (matthew-larson) [reply test/unit/foundation.unit.cpp:376]: Prior to the fix, it crashes in this function.  The same is done in the validation test.  Is there an EXPECT I can use in this situation where is basically passes by not crashing?  I did add a comment to "Expect success" as done previously.

File: src/libkiva/Foundation.cpp (hunk 2 above)
Comment (matthew-larson) [reply src/libkiva/Foundation.cpp]: How is: `showMessage(MSG_WARN, "Surface was modified to be counterclockwise as required in Kiva.");`

Comment (nealkruis) [reply src/libkiva/Foundation.cpp:136]: Sounds good!

Comment (nealkruis) [reply src/libkiva/Foundation.cpp]: Actually, the term "Surface" is a little ambiguous in Kiva. Maybe say "Foundation floor polygon was modified ...".

File: test/unit/foundation.unit.cpp (hunk 1 above)
Comment (nealkruis) [reply test/unit/foundation.unit.cpp:376]: How about: `EXPECT_TRUE(isCounterClockwise(fnd.polygon))`?

**Review by nealkruis — APPROVED** [review]

**Pull request merged**